pkg/frontend/openshiftcluster_listorgetkubernetesobjects_post.go

```diff
@@ -0,0 +1,92 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/v20191231preview"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/dynclient"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+func (f *frontend) postListOrGetKubernetesObjects(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	log.Infof("postListOrGetKubernetesObjects")
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	if len(body) > 0 && !json.Valid(body) {
+		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid and could not be deserialized.")
+		return
+	}
+
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	b, err := f._postListOrGetKubernetesObjects(ctx, r)
+
+	reply(log, w, nil, b, err)
+}
+
+func (f *frontend) _postListOrGetKubernetesObjects(ctx context.Context, r *http.Request) ([]byte, error) {
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	vars := mux.Vars(r)
+
+	_, err := f.validateSubscriptionState(ctx, r.URL.Path, api.SubscriptionStateRegistered)
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err := f.db.OpenShiftClusters.Get(ctx, r.URL.Path)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return nil, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return nil, err
+	}
+
+	if doc.OpenShiftCluster.Properties.ProvisioningState == api.ProvisioningStateCreating ||
```

Comment:
This somehow calls for different construct/helper `State in List` or something similar.